### DISCUSSION TITLE
How to set the optons for 2nd filter base on selected item on 1st filter?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#9663] [Help] [answered]
Hi All.
How to set the optons for 2nd filter base on selected item on 1st filter?

![image](https://github.com/filamentphp/filament/assets/6214964/ecdb9fc0-fde4-40bf-a13d-488cba476cc2)

My table like this :
```
Type_1	| Type_2	| Information
A	| Red		| A-AA
A	| Yellow	| A-AB
A	| Green		| A-AC
A	| Purple	| A-DC
B	| Blue		| B-A1
B	| Red		| B-AA
B	| Green		| B-AB
B	| Orange	| B-AC
```
Type_1 + Type_2 is unique.

```
          SelectFilter::make('type_1')->label('Type-1')
                    ->options(Table1::distinct()
                        ->pluck('type_1', 'type_1'),

          SelectFilter::make('type_2')->label('Type-2')
                    ->options(Table1::where( ??? ) ->distinct()
                        ->pluck('type_2', 'type_2'),
```
What I expected : 

If 1st filter is selected 'A', then the available options on 2nd ar Red,Yellow,Green and Purple
If 1st filter is selected 'B', then the available options on 2nd ar Blue,Red,Green and Oraange

need advice.
thank you

**Comment by sinnbeck** [comment]
First off the first field needs to be set to  ->live(). Secondly you need to set  `->option()` to use a closure. You can use Get to get the state

https://filamentphp.com/docs/3.x/forms/getting-started#dependant-fields

**Comment by devhoussam1998** [comment]
https://www.youtube.com/watch?v=1Sg4bicefNM

**Comment by jigujigu** [comment]
Thanks for advice.
But, this one is SelectFilter component, not the Select Component.
```
          SelectFilter::make('type_1')->label('Type-1')
                    ->options(Table1::distinct()
                        ->pluck('type_1', 'type_1')
                    ->afterStateUpdated(fn (Set $set) => $set('type_2', null))      // Error
```
It will show an error : [Method Filament\Tables\Filters\SelectFilter::afterStateUpdated does not exist]


```
          SelectFilter::make('type_2')->label('Type-2')
                    ->options( 
                           function (Get $get) {
                              $type_1 = $get('type_1');    //Error
                               return (Table1::distinct()-where('Type_1',$type_1));
                            }
                        )->pluck('type_2', 'type_2'),
```
It will show an error : [Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization]

Thanks

**Reply by leandrocfe** [reply, marked as answer]
```php
SelectFilter::make('type_1')
    ->options(Table1::distinct()->pluck('type_1', 'type_1')),

SelectFilter::make('type_2')
    ->options(function ($livewire) {
        return Table1::when(
            $livewire->tableFilters['type_1']['value'],
            fn (Builder $query, $type1) => 
                $query
                ->where('type_1', $type1))
                ->distinct()
                ->pluck('type_2', 'type_2');
    })
```
?

**Reply by MirkoDzudzarLF** [reply]
This is not reactive!
